Automattic/wp-calypso
# Clean up TypeScript support in Storybook

#### Changes proposed in this Pull Request

* Use default TS support instead of a custom webpack (https://storybook.js.org/docs/react/configure/typescript)
* Drop `react-docgen-typescript-loader` support as it only works with Storybook's Info addon that we are not using.
* As a side effect, fixes a few missing peer dependencies:

```
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/packages/components/package.json:46:37: Unmet transitive peer dependency on typescript@*, via react-docgen-typescript-loader@^3.7.2
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/packages/components/package.json:47:16: Unmet transitive peer dependency on typescript@*, via ts-loader@^8.0.8
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/packages/components/package.json:47:16: Unmet transitive peer dependency on webpack@*, via ts-loader@^8.0.8
```

#### Testing instructions

Verify storybooks still work:
* Run `yarn components:storybook:start`, verify it compiles without errors and stories are loaded
* Run `search:storybook:start`, verify it compiles without errors and stories are loaded
